src/widget/wlibrarytableview.cpp

```diff
@@ -149,17 +148,59 @@ void WLibraryTableView::setSelectedClick(bool enable) {
     }
 }
 
-bool WLibraryTableView::event(QEvent* e) {
-    // On FocusIn, with no focused item, select the first track which can then
-    // instantly be loaded to a deck.
-    // This is especially helpful if the table has only one track, which can not
-    // be selected with up/down buttons, either physical or emulated via
-    // [Library],MoveVertical controls. See lp:1808632
-    if (e->type() == QEvent::FocusIn &&
-            model()->rowCount() > 0 &&
-            currentIndex().row() == -1) {
-        selectRow(0);
+void WLibraryTableView::focusInEvent(QFocusEvent* event) {
+    qDebug() << "";
+    qDebug() << "   focusInEvent";
+    qDebug() << "";
+    QTableView::focusInEvent(event);
+
+    if (event->reason() == Qt::TabFocusReason ||
+            event->reason() == Qt::BacktabFocusReason) {
+        // On FocusIn caused by a tab action with no focused item, select the
+        // current or first track which can then instantly be loaded to a deck.
+        // This is especially helpful if the table has only one track, which can
+        // not be selected with up/down buttons, either physical or emulated via
+        // [Library],MoveVertical controls. See lp:1808632
+        if (model()->rowCount() > 0) {
+            if (selectionModel()->hasSelection()) {
+                qDebug() << "";
+                qDebug() << "   selectionModel()->hasSelection()";
+                qDebug() << "";
+
+                DEBUG_ASSERT(!selectionModel()->selectedIndexes().isEmpty());
+                if (!currentIndex().isValid() ||
+                        !selectionModel()->isSelected(currentIndex())) {
+                    if (!currentIndex().isValid()) {
+                        qDebug() << "   !currentIndex().isValid()";
+                    } else {
+                        qDebug() << "   currentIndex().isValid()" << currentIndex().row();
+                    }
+                    if (!selectionModel()->isSelected(currentIndex())) {
+                        qDebug() << "   !selectionModel()->isSelected(currentIndex())";
+                    }
+                    // Refocus the first selected index
+                    setCurrentIndex(selectionModel()->selectedIndexes().first());
+                    selectRow(currentIndex().row());
```

Comment:
I'm not at all sure if all the above checks are necessary, but this last line fixes the issue described above